light-rpc/src/cli.rs

```diff
@@ -0,0 +1,104 @@
+use {
+    clap::{crate_description, crate_name, App, Arg, ArgMatches},
+    solana_clap_utils::input_validators::{is_url, is_url_or_moniker},
+    solana_cli_config::{ConfigInput, CONFIG_FILE},
+    solana_sdk::signature::{read_keypair_file, Keypair},
+    std::{net::SocketAddr, process::exit, time::Duration},
+};
+
+/// Holds the configuration for a single run of the benchmark
+pub struct Config {
+    pub rpc_addr: SocketAddr,
+    pub subsription_port: SocketAddr,
+    pub json_rpc_url: String,
+    pub websocket_url: String,
+}
+
+impl Default for Config {
+    fn default() -> Config {
+        Config {
+            rpc_addr: SocketAddr::from(([127, 0, 0, 1], 8899)),
+            json_rpc_url: ConfigInput::default().json_rpc_url,
+            websocket_url: ConfigInput::default().websocket_url,
+            subsription_port : SocketAddr::from(([127, 0, 0, 1], 8901)),
+        }
+    }
+}
+
+/// Defines and builds the CLI args for a run of the benchmark
+pub fn build_args<'a, 'b>(version: &'b str) -> App<'a, 'b> {
```

Comment:
We can use feature `derive` instead. https://docs.rs/clap/latest/clap/#example.

Reply:
After I commit my code, could you look into it?

Reply:
Sure